### PR TITLE
Add ZSH completion

### PR DESCRIPTION
**Pull request by VorpalBlade**
This is mostly a write of zsh completion from scratch, only partly based on the code in https://github.com/linrunner/TLP/issues/39#issuecomment-36626295. I couldn't reuse most of it since it didn't follow zsh style guideline (use lower case in the help messages) and was also outdated and missing several options.

I believe these completions should be accurate as of 1.5.0 (latest version and latest commit as of writing this).

### CODE REVIEW

**Comment by VorpalBlade** [comment]
One thing I'm not completing (and nor is the bash completion, so I don't consider this a big issue) is config parameter override on the command line.

This could be done, though I prefer to have a hidden command to tlp that lists valid variable names rather than hard coding the list of these in both the zsh and bash completion (and possibly fish completion in the future if someone wants to write that). Completing arguments to those config flags would be more difficult though.

**Comment by linrunner** [comment]
Hi, 

this is really great, 9 years after #39. Give me a little time to examine.

**Comment by linrunner** [comment]
Solves #39

File: completion/zsh/_tlp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    'start:start tlp and apply settings' \
+    'true:alias for bat' \
+    'false:alias for ac' \
+    'bat:apply battery profile and enter manual mode' \
+    'ac:apply AC profile and enter manual mode' \
+    'usb:enable autosuspend for all USB devices except those excluded in the config' \
+    'bayoff:turn off optical bay in UltraBay/MediaBay' \
+    'chargeonce:charge the battery to the stop charge threshold once (ThinkPads only):$subcmd_batteries' \
+    'discharge:force complete discharge of battery (ThinkPads only):$subcmd_batteries' \
+    'setcharge:change charge thresholds temporarily:' \
```
Comment (linrunner) [inline completion/zsh/_tlp:43]: `tlp setcharge 44 88`  doesn't complete battery (4th argument) - refer to the [bash completion](https://github.com/linrunner/TLP/blob/main/tlp.bash_completion#L43).

File: completion/zsh/_tlp-run-on [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#compdef run-on-ac run-on-bat
+
+# Command line is an arbitrary command, so just forward completion to _normal
+_normal
```
Comment (linrunner) [inline completion/zsh/_tlp-run-on:4]: When I press tab twice after entering `run-on-ac` or `run-on-bat` I get 
  
> _normal:4: maximum nested function level reached; increase FUNCNEST?
_normal:4: maximum nested function level reached; increase FUNCNEST?
_normal:4: maximum nested function level reached; increase FUNCNEST?
_normal:4: maximum nested function level reached; increase FUNCNEST?

File: completion/zsh/_tlp-rdw [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#compdef tlp-rdw
+
+_values 'command' \
+    'disable[disable RDW actions]' \
+    'enable[enable RDW actions]'
```
Comment (linrunner) [inline completion/zsh/_tlp-rdw:5]: When I enter (or complete to) `tlp-rdw enable ` (note the trailing space) the completion will continue and happily add any number of _enable_ / _disable_ arguments. There should only be one. Also happens for _disable_ as 1st argument.

File: completion/zsh/_tlp-radio-device [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#compdef bluetooth nfc wifi wwan
+
+_values 'command' \
+    'on[switch device on]' \
+    'off[switch device off]' \
+    'toggle[toggle device state]'
```
Comment (linrunner) [inline completion/zsh/_tlp-radio-device:6]: When I enter (or complete to)  `wifi on ` (note the trailing space) the completion will continue and happily add any number of _on_ / _off_ / _toggle_ arguments. There should only be one. Also happens for _off_ / _toggle_ as 1st argument and for `nfc`, `bluetooth` and `wwan` as well.

**Review by linrunner — CHANGES_REQUESTED** [review]

File: completion/zsh/_tlp (hunk 1 above)
Comment (VorpalBlade) [reply completion/zsh/_tlp:43]: Oops, forgot to handle the arguments for that one. Will fix.

File: completion/zsh/_tlp-rdw (hunk 3 above)
Comment (VorpalBlade) [reply completion/zsh/_tlp-rdw:5]: I learnt something new about how zsh completion works today! Thanks

File: completion/zsh/_tlp-run-on (hunk 2 above)
Comment (VorpalBlade) [reply completion/zsh/_tlp-run-on:4]: huh, it happens for me when I try it in a clean docker container with zsh in it, but not in my fully "kitted out" standard setup. Some further investigation will be required!

File: completion/zsh/_tlp-radio-device (hunk 4 above)
Comment (VorpalBlade) [reply completion/zsh/_tlp-radio-device:6]: same problem as _tlp_rdw. Clearly _values did not quite work how I thought it did.

**Comment by linrunner** [comment]
Right. Thank you very much for your contribution. 

As I have no experience with the zsh, I would be grateful if you could stay tuned in case of issues.

**Comment by VorpalBlade** [comment]
Not a problem, I tag me in any relevant issues and I should be able to respond (though it may take a few days, or weeks later during the summer, when I will be on a completely offline vacation with a backpack and tent).

**Pull request merged**